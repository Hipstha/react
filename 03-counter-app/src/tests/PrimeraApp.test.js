import React from 'react';
import {shallow} from 'enzyme';
import '@testing-library/jest-dom';

import { render } from '@testing-library/react';
import PrimeraAPP from '../PrimeraApp';

describe('Pruebas en <PrimeraApp />', () => {

  // test('Debe mostrar el mensaje "Hola, soy Goku"', () => {

  //   const saludo = 'Hola, soy Goku';

  //   const { getByText } = render( <PrimeraAPP saludo={ saludo } /> );

  //   expect( getByText(saludo) ).toBeInTheDocument();

  // });

  test('debe de mostrar <PrimeraApp /> correctamente', () => {
    const saludo = "Hola, Soy goku";
    const wrapper = shallow( <PrimeraAPP saludo={saludo} /> );

    expect( wrapper ).toMatchSnapshot();

  });

  test('debe de mostrar el subtitulo enviado por props', () => {
    const saludo = "Hola, Soy goku";
    const subtitulo = "Soy un subtitulo"
    const wrapper = shallow( 
      <PrimeraAPP 
        saludo={saludo} 
        subtitulo={subtitulo}
      /> 
    );

    const textoParrafo = wrapper.find('p').text();

    expect( textoParrafo ).toBe( subtitulo );

  })

});