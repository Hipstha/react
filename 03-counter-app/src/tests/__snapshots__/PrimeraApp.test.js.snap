// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> debe de mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola, Soy goku
  </h1>
  <p>
    Default
  </p>
</Fragment>
`;
